tests/src/main/java/net/neoforged/neoforge/debug/capabilities/ItemInventoryTests.java

```diff
@@ -58,40 +68,43 @@ static void init(final TestFramework framework) {
     @TestHolder(description = "Tests that ComponentItemHandler can read and write from a data component")
     public static void testItemInventory(DynamicTest test, RegistrationHelper reg) {
         test.onGameTest(helper -> {
-            ItemStack stack = BACKPACK.toStack();
-            ItemAccess itemAccess = ItemAccess.forStack(stack);
-            // Note: this uses the legacy wrappers, testing the wrappers and that the new ItemAccessItemHandler matches the old ComponentItemHandler.
-            IItemHandler items = IItemHandler.of(itemAccess.getCapability(Capabilities.Item.ITEM));
+            testItemInventory(BACKPACK.toStack(), helper, DataComponents.CONTAINER, ItemContainerContents::getStackInSlot);
+            testItemInventory(BUNDLEPACK.toStack(), helper, DataComponents.BUNDLE_CONTENTS, BundleContents::getItemUnsafe);
+            helper.succeed();
+        });
+    }
 
-            ItemStack storedStick = items.getStackInSlot(STICK_SLOT);
-            helper.assertValueEqual(storedStick.getItem(), Items.STICK, "Default contents should contain a stick at slot " + STICK_SLOT);
+    private static <T> void testItemInventory(ItemStack stack, ExtendedGameTestHelper helper, DataComponentType<T> component, BiFunction<T, Integer, ItemStack> stackGetter) {
```

Comment:
I am not sure that this makes much sense. For the bundle test you should really test various special cases, such as evolution of the slot count as items are inserted, weight of different items, items that cannot be nested, etc...